### PR TITLE
Added the base for CurrentPeriod Entity and tests

### PR DESCRIPTION
**Pull request by sroberts2**
Adds the classes to get the initial current period saved to the database.
Added the relevant test classes.

Helps resolve: SFA-337

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CurrentPeriodServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    @Override
+    public void addLinks(CurrentPeriod rest) {
+
+    }
+
+    @Override
+    public void addKind(CurrentPeriod rest) {
+        rest.setKind("accounts#small-full#current-period");
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CurrentPeriodServiceImpl.java]: Can the kind be built incrementally (like we are planning to do with the links)?

File: src/main/java/uk/gov/companieshouse/api/accounts/model/entity/CurrentPeriodDataEntity.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package uk.gov.companieshouse.api.accounts.model.entity;
+
+import org.springframework.data.mongodb.core.mapping.Field;
+
+public class CurrentPeriodDataEntity extends BaseDataEntity {
+
+    @Field("balance_sheet")
+    BalanceSheetEntity balanceSheetEntity;
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/model/entity/CurrentPeriodDataEntity.java]: Missing private modifier on the variable

File: src/test/java/uk/gov/companieshouse/api/accounts/transformer/CurrentPeriodTransformerTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Test
+    @DisplayName("Tests current period transformer with populated object and validates values returned")
+    public void testTransformerWithPopulatedObject() {
+        BalanceSheet balanceSheet = new BalanceSheet();
+        balanceSheet.setCalledUpShareCapitalNotPaid(5);
+
+        CurrentPeriod currentPeriod = new CurrentPeriod();
+        currentPeriod.setEtag("etag");
+        currentPeriod.setKind("kind");
```
Comment (jsimons1989) [inline src/test/java/uk/gov/companieshouse/api/accounts/transformer/CurrentPeriodTransformerTest.java]: The kind isn't being tested

Comment (sroberts2) [reply src/test/java/uk/gov/companieshouse/api/accounts/transformer/CurrentPeriodTransformerTest.java]: Test added.

**Pull request merged**